Extension/package.json

```diff
@@ -1984,10 +1984,22 @@
                 "description": "%c_cpp.debuggers.cppvsdbg.visualizerFile.description%",
                 "default": ""
               },
-              "externalConsole": {
-                "type": "boolean",
-                "description": "%c_cpp.debuggers.cppvsdbg.externalConsole.description%",
-                "default": false
+              "console": {
+                "type": "string",
+                "enum": [
+                  "internalConsole",
+                  "integratedTerminal",
+                  "externalTerminal",
+                  "newExternalWindow"
+                ],
+                "enumDescriptions": [
+                  "%c_cpp.debuggers.cppvsdbg.console.internalConsole.description%",
+                  "%c_cpp.debuggers.cppvsdbg.console.integratedTerminal.description%",
+                  "%c_cpp.debuggers.cppvsdbg.console.externalTerminal.description%",
+                  "%c_cpp.debuggers.cppvsdbg.console.newExternalWindow.description%"
+                ],
+                "description": "%c_cpp.debuggers.cppvsdbg.console.description%",
+                "default": "internalConsole"
```

Comment:
Is this the best default? It seems like a lot of users hit the cin not working issue. Would another default generate less issues?

Reply:
If not defined, this is the behavior vsdbg uses.  

"default" here is just what is autopropogated if the users tab in `launch.json`

Reply:
To be clear, this is vsdbg's default, but the configuration provider could pick a different default for templates if we think that is better in these scenarios. If I recall correctly, we do use an external console by default in GDB cases? If yes, it might make sense to do the same here. The only downside is that VS Code doesn't switch focus like VS does when a breakpoint is hit. So it can be a bit confusing.